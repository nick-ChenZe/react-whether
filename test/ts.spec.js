import {Whether, Match, Else} from '../dist';
import renderer from 'react-test-renderer';

test('[ts] one branch matched', () => {
    const component = renderer.create(
        <Whether matches={true}>
            <div />
        </Whether>
    );

    expect(component.toJSON()).toMatchSnapshot();
});

test('[ts] one branch with multiple children', () => {
    const component = renderer.create(
        <Whether matches={true}>
            <ul />
            <ol />
        </Whether>
    );

    expect(component.toJSON()).toMatchSnapshot();
});

test('[ts] one branch not matched', () => {
    const component = renderer.create(
        <Whether matches={false}>
            <div />
        </Whether>
    );

    expect(component.toJSON()).toMatchSnapshot();
});

test('[ts] one branch matched with function children', () => {
    const component = renderer.create(
        <Whether matches={true}>
            {() => <div />}
        </Whether>
    );

    expect(component.toJSON()).toMatchSnapshot();
});

test('[ts] if and else when matched', () => {
    const component = renderer.create(
        <Whether matches={true}>
            <div />
            <Else>
                <span />
            </Else>
        </Whether>
    );

    expect(component.toJSON()).toMatchSnapshot();
});

test('[ts] if and else with multiple children matched', () => {
    const component = renderer.create(
        <Whether matches={true}>
            <ul />
            <ol />
            <Else>
                <span />
            </Else>
        </Whether>
    );

    expect(component.toJSON()).toMatchSnapshot();
});

test('[ts] if and else when not matched', () => {
    const component = renderer.create(
        <Whether matches={false}>
            <div />
            <Else>
                <span />
            </Else>
        </Whether>
    );

    expect(component.toJSON()).toMatchSnapshot();
});

test('[ts] else with multiple children', () => {
    const component = renderer.create(
        <Whether matches={true}>
            <div />
            <Else>
                <ul />
                <ol />
            </Else>
        </Whether>
    );

    expect(component.toJSON()).toMatchSnapshot();
});

test('[ts] if and else when not matched with function children', () => {
    const component = renderer.create(
        <Whether matches={false}>
            <div />
            <Else>
                {() => <span />}
            </Else>
        </Whether>
    );

    expect(component.toJSON()).toMatchSnapshot();
});

test('[ts] multiple match', () => {
    const component = renderer.create(
        <Whether context={3}>
            <Match selector={i => i < 1}>
                <span />
            </Match>
            <Match selector={i => i > 2}>
                <div />
            </Match>
        </Whether>
    );

    expect(component.toJSON()).toMatchSnapshot();
});

test('[ts] match with multiple children', () => {
    const component = renderer.create(
        <Whether context={3}>
            <Match selector={i => i < 1}>
                <span />
            </Match>
            <Match selector={i => i > 2}>
                <ul />
                <ol />
            </Match>
        </Whether>
    );

    expect(component.toJSON()).toMatchSnapshot();
});

test('[ts] multiple match when none matched', () => {
    const component = renderer.create(
        <Whether context={3}>
            <Match selector={i => i < 1}>
                <div />
            </Match>
            <Match selector={i => i < 2}>
                <span />
            </Match>
        </Whether>
    );

    expect(component.toJSON()).toMatchSnapshot();
});

test('[ts] multiple match with else matched', () => {
    const component = renderer.create(
        <Whether context={3}>
            <Match selector={i => i < 1}>
                <div />
            </Match>
            <Else>
                <span />
            </Else>
        </Whether>
    );

    expect(component.toJSON()).toMatchSnapshot();
});

test('[ts] match with function children', () => {
    const component = renderer.create(
        <Whether context={3}>
            <Match selector={i => i > 1}>
                {() => <div />}
            </Match>
            <Else>
                {() => <span />}
            </Else>
        </Whether>
    );

    expect(component.toJSON()).toMatchSnapshot();
});

test('[ts] else with function children', () => {
    const component = renderer.create(
        <Whether context={3}>
            <Match selector={i => i < 1}>
                {() => <div />}
            </Match>
            <Else>
                {() => <span />}
            </Else>
        </Whether>
    );

    expect(component.toJSON()).toMatchSnapshot();
});
