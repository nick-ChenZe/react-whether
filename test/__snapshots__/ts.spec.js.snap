// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[ts] else with function children 1`] = `<span />`;

exports[`[ts] else with multiple children 1`] = `<div />`;

exports[`[ts] if and else when matched 1`] = `<div />`;

exports[`[ts] if and else when not matched 1`] = `<span />`;

exports[`[ts] if and else when not matched with function children 1`] = `<span />`;

exports[`[ts] if and else with multiple children matched 1`] = `
Array [
  <ul />,
  <ol />,
]
`;

exports[`[ts] match with function children 1`] = `<div />`;

exports[`[ts] match with multiple children 1`] = `
Array [
  <ul />,
  <ol />,
]
`;

exports[`[ts] multiple match 1`] = `<div />`;

exports[`[ts] multiple match when none matched 1`] = `null`;

exports[`[ts] multiple match with else matched 1`] = `<span />`;

exports[`[ts] one branch matched 1`] = `<div />`;

exports[`[ts] one branch matched with function children 1`] = `<div />`;

exports[`[ts] one branch not matched 1`] = `null`;

exports[`[ts] one branch with multiple children 1`] = `
Array [
  <ul />,
  <ol />,
]
`;
